pytradfri/resource.py

```diff
@@ -1,49 +1,65 @@
 """Resources for devices."""
+from __future__ import annotations
+
 from datetime import datetime
+from typing import Any, Dict, List, Union, cast
 
-from .command import Command
+from .command import TYPE_ERR_CB, TYPE_PROCESS_RESULT_CB, Command
 from .const import ATTR_CREATED_AT, ATTR_ID, ATTR_NAME
 
+# type alias
+TYPE_RAW = Dict[str, Union[str, int, List[Dict[str, Union[str, int]]]]]
+TYPE_RAW_SIMPLE = Dict[str, Union[str, int]]
+TYPE_RAW_LIST = Dict[str, List[Dict[str, Union[str, int]]]]
```

Comment:
Something to consider for the future is to use a dataclass or TypedDict to represent the RAW data. I'd prefer dataclass without looking into any details yet.